main/lsp/updates.cc

```diff
@@ -156,19 +155,13 @@ void tryApplyDefLocSaver(const core::GlobalState &gs, vector<ast::ParsedFile> &i
     }
 }
 
-LSPLoop::TypecheckRun LSPLoop::runSlowPath(const vector<shared_ptr<core::File>> &changedFiles) {
+LSPLoop::TypecheckRun LSPLoop::runSlowPath() {
```

Comment:
The reason this was historically done is to have `runSlowPath` invalidate everything, including potential bugs in how trees were indexed.

While I don't believe we had bugs like this in practice, I'd like to call this out as if we ever do, it'll make it hard to find the

Reply:
I don't follow. `tryFastPath` calls `updateFile`, which unconditionally indexes the changed files with `initialGS` and invalidates the old `ParsedFile`s.

Reply:
`updateFile` resyncs the `indexed` tree to match the file structure.
This makes sure that if there were bugs and something forgot to update it, it gets cleared

Reply:
IRLed:

`runSlowPath` re-indexed the changed files in *the latest file update* before running the slow path. If there were other file updates that occurred on the fast path, they would not be re-indexed.

So if changes 2,3,4 ran on the fast path, and 5 triggers slow path, `runSlowPath` only re-indexes the changes in 5.

To get the guarantee, we could modify the logic to re-index all files changed on fast path. This is possible now that I track `ParseFile`s from fast path indexes separately.

Reply:
I think it's peaceful to land this, since it's not guaranteeing what Dmitry expected that it did. I'll open a PR shortly that contains an assertion that tries to uphold this guarantee.